src/librustc_mir/transform/const_prop.rs

```diff
@@ -553,6 +553,45 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 }
             }
 
+            Rvalue::Cast(CastKind::Misc, op, ty) => {
```

Comment:
The function is becoming rather large... can we extract the whole / parts of the branch?

Reply:
Sure, pulling out the various checks significantly reduces the size of the function.